packages/salesforcedx-vscode-core/src/commands/forceStopApexDebugLogging.ts

```diff
@@ -65,16 +65,12 @@ export async function restoreDebugLevels(): Promise<void> {
   }
 }
 
-function getRestoreLevelsCommand(): Command {
+function deleteTraceFlag(): Command {
   return new SfdxCommandBuilder()
     .withDescription(nls.localize('force_stop_apex_debug_logging'))
-    .withArg('force:data:record:update')
-    .withFlag('--sobjecttype', 'DebugLevel')
-    .withFlag('--sobjectid', developerLogTraceFlag.getDebugLevelId())
-    .withFlag(
-      '--values',
-      `ApexCode=${developerLogTraceFlag.getPrevApexCodeDebugLevel()} Visualforce=${developerLogTraceFlag.getPrevVFCodeDebugLevel()}`
-    )
+    .withArg('force:data:record:delete')
+    .withFlag('--sobjecttype', 'TraceFlag')
+    .withFlag('--sobjectid', developerLogTraceFlag.getTraceFlagId()!)
```

Comment:
Same with the explicit `!` here.

Reply:
Similar to the UpdateTraceFlag, this is called after the query executor and so the id should be set before calling. Otherwise it should surface an error to the user